src/test/java/it/pagopa/pn/templatesengine/service/TemplateServiceTest.java

```diff
@@ -41,34 +42,38 @@ public void setup() {
     }
 
     @Test
-    void executeTextTemplate() {
+    void executeTextTemplate_Success() {
         // Arrange
         MailVerificationCodeBody emailbody = new MailVerificationCodeBody();
         emailbody.setVerificationCode("VerificationCode");
 
         // Act & Assert
-        Mono<byte[]> result = Assertions.assertDoesNotThrow(() ->
-                templateService.executePdfTemplate(TemplatesEnum.MAIL_VERIFICATION_CODE_BODY, LANGUAGE, Mono.just(emailbody)));
+        Mono<String> result = Assertions.assertDoesNotThrow(() ->
+                templateService.executeTextTemplate(TemplatesEnum.MAIL_VERIFICATION_CODE_BODY, LANGUAGE, Mono.just(emailbody)));
 
         StepVerifier.create(result)
-                .expectNextCount(1)
+                .assertNext(r -> Assertions.assertTrue(r.contains("VerificationCode"),
+                            "Result should include the verification code"))
                 .verifyComplete();
     }
 
     @Test
     void executePdfTemplate_Success() {
         // Arrange
-        MailVerificationCodeBody emailbody = new MailVerificationCodeBody();
-        emailbody.setVerificationCode("VerificationCode");
+        var model = new MalfunctionLegalFact()
+                .startDate("TEST_startDate")
+                .timeReferenceStartDate("TEST_timeReferenceStartDate")
+                .endDate("TEST_endDate")
+                .timeReferenceEndDate("TEST_timeReferenceEndDate");
 
         // Act & Assert
-        Mono<String> result = Assertions.assertDoesNotThrow(() ->
-                templateService.executeTextTemplate(TemplatesEnum.MAIL_VERIFICATION_CODE_BODY, LANGUAGE, Mono.just(emailbody)));
+        Mono<byte[]> result = Assertions.assertDoesNotThrow(() ->
+                templateService.executePdfTemplate(TemplatesEnum.MALFUNCTION_LEGAL_FACT, LANGUAGE, Mono.just(model)));
 
         StepVerifier.create(result)
-                .assertNext(actualResult -> {
-                    Assertions.assertTrue(actualResult.contains("VerificationCode"),
-                            "Result should include the verification code");
+                .assertNext(r -> {
+                    Assertions.assertNotNull(r);
+                    Assertions.assertEquals(r.length, 60560);
```

Comment:
Is it possible to use other checks instead of length? in this way if copy changes (for examples), test will fail.

Reply:
If copy changes QA team will check manually the PDF